src/aleph/http/client.clj

```diff
@@ -609,6 +614,16 @@
        ([_ ctx]
          (let [ch (.channel ctx)]
            (reset! in (netty/buffered-source ch (constantly 1) 16))
+
+           ;; Start handshake timeout timer
+           (reset! timeout-task
```

Comment:
I was trying to figure out why you used the `.in` accessor here, and realized that `manifold.time/in` doesn't allow it to be cancelled.  That's my bad, I'm going to merge these changes in and then fix the underlying method so we don't have to reach under the covers.